components/base/sensorcontrolled/sensorcontrolled.go

```diff
@@ -279,3 +280,47 @@ func (sb *sensorBase) Close(ctx context.Context) error {
 	sb.activeBackgroundWorkers.Wait()
 	return nil
 }
+
+// determineHeadingFunc determines which movement sensor endpoint should be used for control.
+// The priority is Orientation -> Heading -> No heading control.
+func (sb *sensorBase) determineHeadingFunc(ctx context.Context) {
+	switch {
+	case sb.orientation != nil:
```

Comment:
[q] do we need orientation anymore? or just the heading function?

Reply:
I think its fine to keep orientation(and compass heading) as part of the sensorcontrolled state for now because theres some logic that gets simplified by having it. I could refactor some parts of the code to remove it, would just need to update the headingfunc to return another value that lets me know spin is not supported